src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1594,6 +1596,101 @@ private static unsafe nuint NarrowUtf16ToAscii_Sse2(char* pUtf16Buffer, byte* pA
             goto Finish;
         }
 
+        private static unsafe nuint NarrowUtf16ToAscii_AdvSimd(char* pUtf16Buffer, byte* pAsciiBuffer, nuint elementCount)
+        {
+            // This meethod follows similar logic as the NarrowUtf16ToAscii_Sse2 method above.
+
+            uint SizeOfVector128 = (uint)Unsafe.SizeOf<Vector128<byte>>();
+            nuint MaskOfAllBitsInVector128 = (nuint)(SizeOfVector128 - 1);
+
+            Debug.Assert(AdvSimd.Arm64.IsSupported);
+            Debug.Assert(BitConverter.IsLittleEndian);
+            Debug.Assert(elementCount >= 2 * SizeOfVector128);
+
+            // Process the first chunk of 16 bytes
+
+            ref ushort utf16Buffer = ref *(ushort*)pUtf16Buffer;
+            ref byte asciiBuffer = ref *pAsciiBuffer;
+            Vector128<ushort> utf16VectorFirst = Vector128.LoadUnsafe(ref utf16Buffer);
+            Vector128<ushort> maxChars = AdvSimd.Arm64.MaxPairwise(utf16VectorFirst, utf16VectorFirst);
```

Comment:
This and the corresponding `TestZ` logic in `Sse2` could be just `(x & y) != Zero`.

It won't be quite as efficient as what we have right now, but its a simple pattern we could recognize and optimize in the JIT.